### ISSUE TITLE
Failed to create instance of v7 engine on Windows (VS2013,VS2015)

### ISSUE THREAD

**Issue by kkutsner** [cesanta/v7#551]
I did several exercises against 2.0 release and master branch of v7 on OS X successfully and then moved to Windows...

Damn! I spent an hour or so to figure out what was going. Essentially, I just created VS2013/VS2015 C++ project, put `v7.h` header and `v7.c` source files into project and set `V7_EXE` and `V7_ALLOW_ARGLESS_MAIN` preprocessor definitions. When running the executable it constantly triggered abort message here:

```C
V7_PRIVATE void bcode_op_lit(struct bcode_builder *bbuilder, enum opcode op,
                             lit_t lit) {
  bcode_op(bbuilder, op);

  switch (lit.mode) {
...
    default:
      /* invalid literal mode */
      abort(); // <------------------------------!!!!
      break;
  }
}
```

The problem was using `lit.mode` in switch expression. Since it is defined as below:
```C
typedef struct {
...
  enum lit_mode mode : 1; // <------ Note: specific set of bits is used!!!!
} lit_t;
```
The remain bits of `mode` field may contain trash. So I just changed switch expression to 
```C
  switch (lit.mode & 1) {
```
and everything started work now as expected.

Not sure if you aware of this issue, but I was really upset.... May be I have very specific environment (OSX+Parrallels+Windows 10) so the issue started appearing... I don't know...

**Comment by damospiderman** [comment]
I was also having this issue and the solution above fixed the issue. Thanks

**Comment by mmikulicic** [comment]
wow, that's ... surprising

I don't think it's garbage, I think it gets sign extended. 
(PS: yes I checked with `wine` and `vc6` and indeed LIT_MODE__INLINED turns into -1)

```
6.7.2.2 Enumeration specifiers, paragraph 4

Each enumerated type shall be compatible with char, a signed integer type, or an unsigned integer type. The choice of type is implementation-defined, but shall be capable of representing the values of all the members of the enumeration. The enumerated type is incomplete until immediately after the } that terminates the list of enumerator declarations, and complete thereafter.
```

Let's just increase the size of the bitfield since that should be changed anyway if the enum grows.

**Comment by mmikulicic** [comment]
merged in 8a8bc42dfb4dbc6a6ad677b148c3e9f5cdc2b148
